reactioncommerce/reaction
# Default settings for parcel size

Resolves #3019   
Impact: **minor**  
Type: **chore**

## Issue
When there was no `parcel` present in the `cart` the shipping APIs would not be able to calculate the tax and would err. This case would occur when the Admin hasn't added the dimensions of the product.

## Solution
1. Create a form where the `Admin` can enter the default dimensions that will can be used as a fallback when product dimensions are missing.
1. Create a default for the "default parcel size".

## Breaking changes
None

## Testing
1. Do a `reaction reset -y`
1. Enable shippo.
1. Create and publish a product, but don't add it's dimensions.
1. Buy the product with shippo. Check in the database that `Order.item[0].parcel` = default dimension.
1. Go to shipping -> Parcel Size and change the values
1. Repeat above steps and check the same results.
1. Create a new marketplace shop, enable it.
1. Invite a user to it and log in as that user.
1. Repeat above steps and check the same results. (the parcel settings should get copied over from primary shop to new shop.)

Behavior for old users ->
1. Until the old shop's Admin update the `Parcel Size` settings, the behavior would be same as that was before for them, i.e  the shipping APIs would not be able to calculate the tax and would err if product dimensions are not there.
